### PR TITLE
Changes for using action parameter of orkestra workflow executor

### PR DESCRIPTION
**Pull request by davutkarabay**
The action "Install" and "Delete" will be used for forward and reverse workflows respectively.

Should go with orkestra-workflow-executor [PR](https://github.com/Azure/orkestra-workflow-executor/pull/1).

### CODE REVIEW

**Review by jonathan-innis — COMMENTED** [review]
Just to clarify, because I think I missed some of the conversation around it. Did we determine that the current workflow does not wait for the removal of the finalizer?

File: pkg/workflow/argo.go [hunk 1]
```diff
@@ -329,7 +329,7 @@ func (a *argo) generateReverseWorkflow(ctx context.Context, l logr.Logger, nodes
 
 	updateWorkflowTemplates(a.rwf, entry)
 
-	updateWorkflowTemplates(a.rwf, defaultReverseExecutor())
+	updateWorkflowTemplates(a.rwf, defaultExecutor(helmReleaseReverseExecutor, "delete"))
```
Comment (nitishm) [inline pkg/workflow/argo.go:332]: Let's define the action strings ("install" & "delete") as `type ExecutorAction string`  and create `const`s for each one.

**Review by nitishm — CHANGES_REQUESTED** [review]
Nit

**Comment by nitishm** [comment]
@jonathan-innis No rush since you're OOF - but if you get bored can you take a look at why the e2e tests are failing. Since this is a pretty big change moving from the install only version of the executor image to the new version with both install and delete, it is possible that things might be broken or some incorrect assumptions might have been made.
@davutkarabay Can you compare the install code line-by-line and see if anything has changed in terms of underlying functions or timing delays etc since version v0.2.0?

**Pull request merged**